canonical-web-and-design/maas-ui
# feat(ui): build machine power configuration form

## Done

- Built "Power configuration" section of the config tab
- Changed links to point to the React version of the config tab
- Updated `FormikForm`'s `onCancel` prop so you can give it the entire formik context as an argument and access things like `initialValues` and `resetForm`. I've used it in the MachineForm and PowerForm to reset the form when "Cancel" is pressed.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Create a machine manually (i.e. not in a pod) and go to its details page
- Click "Configuration" and check that it takes you to the React configuration screen
- Check that the power details are initialised correctly
- Click the "Edit" button next to "Power configuration"
- Check that the select is enabled and you can change the power type and power parameters
- Change a few values then click "Cancel" and check that the values are reverted to the originals
- Enter an invalid value (e.g "1" for an ip address) and submit and check that en error is displayed
- Check that you can successfully update the power config with valid values
- Go to the config page for a machine that's in a pod
- Check that the select is always disabled, and you only see fields that are scoped to individual nodes

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2223
